### ISSUE TITLE
Arabic display: don't understand

### ISSUE THREAD

**Issue by jcowey** [papyri/navigator#118]
https://papyri.info/ddbdp/p.heid;11;459

If you load that and then refresh it, for a brief fraction of a second line 4 tries to display as one line.
It is one line:

https://github.com/papyri/idp.data/blob/e9c52a8fed3eca2bc0eef4c147a25f25bbf86195/DDB_EpiDoc_XML/p.heid/p.heid.11/p.heid.11.459.xml#L55

If you look at the history / blame of the XML file you will see that I closed the `foreign` tag immediately after the Arabic entry then allowed `<space extent="unknown" unit="character"/>` before opening the `foreign` tag again. (If one leave `<space extent="unknown" unit="character"/>` within the `foreign` tag the parts of the Arabic line are inverted, which produced nonsense - all to do with the LTR / RTL idiosyncrasies)

I thought I had got 

https://github.com/papyri/idp.data/blob/e9c52a8fed3eca2bc0eef4c147a25f25bbf86195/DDB_EpiDoc_XML/p.heid/p.heid.11/p.heid.11.459.xml#L55

correct (following the example of line 1, which is correct), but the line splits and jumps to produce two half lines.

Would you please try to solve the problem.

**Comment by ryanfb** [comment]
For reference, if you view-source the HTML it looks correct:
```html
<br id="al4"><span class="lang" lang="ar" dir="rtl">[محمد رسول الـله]</span> vac. ? <span class="lang" lang="ar" dir="rtl">لا اله الا الـله وحده</span>
```
However, if I "Inspect Element" on the page after it's loaded (checked in Safari/Firefox), there's another span somehow inserted which is causing this issue:
```html
<br id="al4">
<span style="display:inline-block;width:337px;"> </span>
<span class="lang" lang="ar" dir="rtl">[محمد رسول الـله]</span>
 vac. ? 
<span class="lang" lang="ar" dir="rtl">لا اله الا الـله وحده</span>
```

**Comment by hcayless** [comment]
There's some code to try to format Arabic next to LTR text, and I think it's getting confused by the vac. Shouldn't be too hard to fix.

**Comment by hcayless** [comment]
Ok, I've got a variety of concerns here, which I'll try to enumerate:

1. I'm worried about aspects of the encoding. The text must start with "بسم الـله" (bism Allah—in the name of God), right? But in the source you have
```xml
<lb n="1"/><foreign xml:lang="ar"><supplied reason="lost">الرحمن الرحيم</supplied></foreign> <space extent="unknown" unit="character"/> <foreign xml:lang="ar"> بسم الـله</foreign>
```
So the line beginning is the third element in the line, where it should be first. I know the order looks right, but it's backwards.

2. I think this is a single stretch of Arabic with a space in the middle, not two separate stretches, so a better encoding would be:
```xml
<lb n="1"/><foreign xml:lang="ar">بسم الـله <space extent="unknown" unit="character"/> <supplied reason="lost">الرحمن الرحيم</supplied></foreign>
```
(I hope I haven't butchered that. Arabic and raw XML don't mix well)

3. I'm going to suggest a change to the EpiDoc Stylesheets so that `<space>` gets turned into `<span dir="ltr">vac.</span>`, etc., so that display weirdness doesn't happen when the browser decides RTL is the dominant direction (else we'll get "? .vac" instead of "vac. ?". That would make the suggestion above work properly.
4. I think we need some alternatives to `<foreign>` for this kind of document. In these things I think it's hard to argue that there's really a "dominant" language. Greek seems to come first here, but there are similar ones where it's the opposite. I think we should use something like `<seg xml:lang="ar">` for cases where there are stretches of one language or another where `<foreign>` might not be appropriate. That would entail some Leiden+ work too.
5. The bug you've noticed is the page's Javascript trying to right-align Arabic text and it's confused because of the markup. A single `<foreign>` tag for the whole line would sidestep the problem in this case. But it's a bit experimental, so I can't guarantee it will always work correctly. 
6. It might be helpful to set up an Oxygen stylesheet for use with these texts and edit them in Author mode, because using Text mode with Arabic is a nightmare. This is not a solvable problem in Text mode, it's just due to the way the Unicode Bidirectional Algorithm works.

I'm going to do 3 right away. I can put it in a branch and we can use that until the pull request I'll do gets accepted. I'm happy to work with you to get the source right. It's not easy.

**Comment by jcowey** [comment]
3. does seem good. 

I agree that the `space` is "an Arabic space". I did what I did to try not to get caught up in what I thought might become a fankle in github when fiddling with LTR / RTL (jumping that I find difficult to control and know what is going on)

I think that 2. is correct. bismallah at the beginning of the XML followed by `space` followed by `supplied`, which gets inverted on display.

You are right it is not easy. Yes, to improve the working of this would be good. Lajos Berkes entered the Arabic in the editor, where it is likewise easy to get in a fankle.
